Isolate course blocks

renatho opened the pull request:
Part of https://github.com/Automattic/sensei/issues/3821

### Changes proposed in this Pull Request

* This PR purpose isolating the course blocks, so we can introduce the lesson blocks separately.
* This also contains some small cleanups and fixes.

### Testing instructions

* Create a course with blocks, and make sure it continues working properly.

yscik commented on assets/blocks/blocks-setup.js:
```diff
@@ -0,0 +1,16 @@
+
+import { SenseiIcon } from '../icons';
+
+const blocksSetup = ( blocks ) => {
```
How about something like `registerSenseiBlocks`?

renatho replied:
I used a more generic name because we were also setting the category icon, and we can to other stuff there, but I think this name works well too.

Updated in https://github.com/Automattic/sensei/pull/3885/commits/19ed087d9f7f780aaebefa0c3e2126211cbb362a

yscik commented on includes/blocks/class-sensei-course-blocks.php:
```diff
@@ -86,7 +83,7 @@ public function enqueue_block_editor_assets() {
 		}
 
-		Sensei()->assets->enqueue( 'sensei-blocks', 'blocks/index.js', [], true );
+		Sensei()->assets->enqueue( 'sensei-course-blocks', 'blocks/course-blocks.js', [], true );
```
We could go with `single-course` for the name, if it makes sense that these are grouped based on where they need to be loaded. The CSS files are named like that right now.

renatho replied:
Thank you for your review, @yscik!

I got your point, and I think it makes sense. But I think we could make this change in a future PR (maybe when we need that), just to avoid conflicts with this now: https://github.com/Automattic/sensei/pull/3885#issuecomment-758826879

It's because we'd need also to move this file out of the `blocks` folder to make sense.

gkaragia commented:
Hey @renatho I have done some similar work in #3849. Is it possible to consolidate these two or prioritize the other?

renatho commented:
> Hey @renatho I have done some similar work in #3849. Is it possible to consolidate these two or prioritize the other?

Hey @gkaragia! Thank you for looking! As we talked, I took a look, and I did some things to make it easier:

- I removed my change about the JS file, and I cherry-pick yours: https://github.com/Automattic/sensei/pull/3885/commits/12f02ffdc65f5e4ddfe31294c26373d1159737c5
- About the `$sensei` variable in some classes, I think we implemented the same (the only difference is that I removed a space before a period in a comment). I tried to cherry-pick yours, but yours had some other things, so I kept it untouched, but considering the change is only the space, I believe we won't have problems while merging. :)

Anything else, let me know.

gkaragia reviewed (approved):
Looks good, works well!

jom reviewed (approved):
Looks like good changes.
